lib/ansible/plugins/callback/foreman.py

```diff
@@ -0,0 +1,211 @@
+# -*- coding: utf-8 -*-
+# (C) 2015, 2016 Daniel Lobato <elobatocs@gmail.com>
+#     2016 Guido Günther <agx@sigxcpu.org>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+"""
+This callback will report facts and reports to Foreman https://theforeman.org/
+
+It makes use of the following environment variables:
+
+  FOREMAN_URL: URL to the Foreman server
+  FOREMAN_SSL_CERT: X509 certificate to authenticate to Foreman if
+    https is used
+  FOREMAN_SSL_KEY: the corresponding private key
+  FOREMAN_SSL_VERIFY: wether to verify the Foreman certificate
+    It can be set to '1' to verify SSL certificates using the
+    installed CAs or to a path pointing to a CA bundle. Set to '0'
+    to disable certificate checking.
+
+"""
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import os
+from datetime import datetime
+from collections import defaultdict
+import json
+import time
+
+try:
+    import requests
+    HAS_REQUESTS = True
+except ImportError:
+    HAS_REQUESTS = False
+
+from ansible.plugins.callback import CallbackBase
+
+FOREMAN_URL = os.getenv('FOREMAN_URL', "http://localhost:3000")
```

Comment:
Move `FOREMAN_URL` and the other variables that follow inside the `CallbackModule` class.